core/src/main/java/com/flipkart/varadhi/core/cluster/MessageChannel.java

```diff
@@ -0,0 +1,50 @@
+package com.flipkart.varadhi.core.cluster;
+
+
+import com.flipkart.varadhi.core.cluster.messages.*;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * This is subset of methods from vertx.EventBus which are node specific. Instead of using address concept, we will
+ * rely on path concept from http. address is not needed as this class is meant to represent a message channel to a specific
+ * node.
+ * publish methods are fire & forget.
+ * send methods don't have any response, but their delivery can be tracked using the future.
+ * request methods are for traditional request-response pattern.
+ *
+ * TODO: whether to use the ClusterMessage class. It is unlikely we will use reply-to-reply feature.
+ */
+public interface MessageChannel {
+
+    void publish(String path, ClusterMessage msg);
+
+    CompletableFuture<Void> send(String path, ClusterMessage msg);
+
+    CompletableFuture<ResponseMessage> request(String path, ClusterMessage msg);
+
+    //    void addMessageHandler(String path, MessageHandler messageHandler);
+    <E extends ClusterMessage> void register(String address, Class<E> messageClazz, SendHandler<E> handler);
```

Comment:
why is address needed here?
as per doc & send methods, it shouldn't require address param.